modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,339 @@
+// @Notice this part of the code is in the coalescing PR https://github.com/ngrx/platform/pull/2456
+// import { generateFrames } from '../projections/generateFrames';
+// import { coalesce } from '../operators/coalesce';
+import { defer, from, MonoTypeOperatorFunction, Observable } from 'rxjs';
+import {
+  ChangeDetectorRef,
+  NgZone,
+  ɵdetectChanges as detectChanges,
+  ɵmarkDirty as markDirty,
+} from '@angular/core';
+import { hasZone, isIvy } from '../utils';
```

Comment:
@SanderElias suggests to use the window object to determine if zone is present or not

Reply:
Sometimes we pass the `'noop'` `ngZone` option but forget to remove `zone.js` imports from `polyfills.ts` 🤷‍♂️ In this case, we should be checking for NoopNgZone, not just global Zone.js objects and monkeypatches.

Reply:
Good Point!

Reply:
@LayZeeDK 

So there are 2 things:
- zone.js is used and patches the global API, or not
- Angulars NgZone  service is of type `NgZone` or `NgNoopZone`

I thought if there are valid cases where `zone.js` patches the global API and Angular is NOT using `NgZone` but `NgNoopZone`. 

I found Angular elements as a valid example where you can write the element with `NgNoopZone` and the consumer with `NgZone`. 

**So it requires me to always check for the ng zone services instance. Right?**

As I changed this logic 2 times now I thought I make it nice first this time.

I pushed 2 new files:
- zone-checks.ts
- view-engine-checks.ts

The detailed documentation is present now for all the different cases and checks.

Reply:
@LayZeeDK
I rethought the whole thing. After different experiments and tests I concluded a really nice thing!

The really only thing I have to check is if the PROMISE API is untouched or not (If we evaluate a safe usage of animationFrame that it would be that API).

The rest can now be used in both ZoneLess and ZoneFull without reflecting it in the code. I basically kicked out ALL zone related checks everywhere but the creation of Promise that is here for coalescing.

This refactoring also included:
- get rid of passing the component
- most probable get ird of _lView (if _lView  === cdRef.context across class and pipe/directive in the template)
- getting rid if generateFrages RxJS operator
- get rid of All zone related tests but the one for the Promise

Thanks a lot for making me think the 10th time on that problem !!!

Reply:
Conclusion:
The only thing where I have to deal with `zone.js` is in the coalescing logic and the only API interesting to check is the `Promise` API.

Reply:
Let me know your thoughts